loading.bs

```diff
@@ -988,6 +1057,90 @@ A [=request=] |browserRequest| <dfn>matches the stored exchange</dfn>
     Behavior=] returning a list of lists.
 1. Return "match".
 
+<h3 algorithm id="create-a-new-report">Create a new signed exchange report</h3>
+To <dfn>create a new signed exchange report</dfn> with |request| and
+|actualResponse|, the UA MUST:
+
+1. Let |report| be a new [=signed exchange report=] struct.
+1. Set |report|'s [=signed exchange report/outer request=] to |request|.
+1. Set |report|'s [=signed exchange report/outer response=] to |actualResponse|.
+1. Set |report|'s [=signed exchange report/server IP=] to the IP address of the
+    server from which the user agent received the |actualResponse|, if
+    available.
+1. Return |report|.
+
+<h3 algorithm id="wait-and-queue-a-report">Wait and queue a report</h3>
+To <dfn>wait and queue a report for</dfn> |parsedExchange| and |report|, the UA
+MUST:
+
+1. Wait until |parsedExchange|'s [=response/body=]'s [=body/stream=] is
+    [=ReadableStream/closed=] or [=ReadableStream/errored=].
+1. If |parsedExchange|'s [=response/body=]'s [=body/stream=] is
+    [=ReadableStream/closed=], run [=queue a signed exchange report=] |report|
+    with "[=signed exchange report/ok=]" as the result and abort these steps.
+1. If |parsedExchange|'s [=response/body=]'s [=body/stream=] is
+    [=ReadableStream/errored=], run [=queue a signed exchange report=] |report|
+    with "[=signed exchange report/mi_error=]" as the result.
+
+<h3 algorithm id="queue-report">Queuing signed exchange report</h3>
+
+To <dfn>queue a signed exchange report</dfn> |report| with |result| as the
+result, the UA MUST:
+
+1. Set |report|'s [=signed exchange report/result=] to |result|.
+
+1. Let |report body| and |policy| be the result of
+    [=generate a network error report=]</a> with |report|'s
+    [=signed exchange report/outer request=]. If the result is null, abort
+    these steps.
+
+1. If |report body|'s `"type"` is `"dns.address_changed"`, abort these steps.
+
+    Note: This means that the NEL report was downgraded because the IP addresses
+    of the server and the |policy| don't match. In this case, UA have called
+    [=deliver a network report=] algorithm with the error report while handling
+    the response. So we don't need to send the same error report while
+    processing the response as a signed exchange.
+
+1. Add a new property `"sxg"` to |report body| with a new ECMAScript object with
+     the following properties:
+
+    * `outer_url`: The [=URL serializer|serialization=] of |report|'s
+          [=signed exchange report/outer request=]'s [=request/url=].
+    * `inner_url`: The [=URL serializer|serialization=] of |report|'s
+          [=signed exchange report/inner URL=].
+    * `cert_url`: The [=sequence type=] of the result of
+          [=URL serializer|serialization=] of each element of |report|'s
+          [=signed exchange report/cert URL list=].
+
+1. Set |report body|'s `"phase"` to `"sxg"`.
+
+1. Set |report body|'s `"type"` to the result of concatenating a string `"sxg."`
```

Comment:
My instinct is that the `type` should just be `"ok"` in the success case, and not `"sxg.ok"`, to match other successful loads, but I defer to @igrigorik or @dcreager.

Reply:
Agreed.  `ok` should be the standard value for "everything worked as expected".  I like the `sxg.` prefix for any non-success results.

Reply:
Done.